src/commands.ts

```diff
@@ -248,6 +248,147 @@ export async function createNewProject(ctx: WorkspaceContext): Promise<void> {
+    // The context key `swift.addPackageRefactoringAvailable` only works if the extension has been
+    // activated. As such, we also have to allow this command to run when no workspace is
+    // active. Show an error to the user if the command is unavailable.
+    if (!ctx.toolchain.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))) {
+        vscode.window.showErrorMessage(
+            "Browsing the package index is only available starting in swift version 6.0.0."
+        );
+        return;
+    }
+
+    const panel = vscode.window.createWebviewPanel(
+        "swiftPackageIndex",
+        "Swift Package Index",
+        vscode.ViewColumn.One,
+        {
+            retainContextWhenHidden: true,
+            enableScripts: true,
+        }
+    );
+
+    panel.webview.onDidReceiveMessage(
+        message => {
+            switch (message.command) {
+                case "addSwiftPackage":
+                    addPackageDependency(ctx, message.url, message.version);
+                    return;
+                default:
+                    // Log message
+                    vscode.window.showErrorMessage(`An error occurred when adding packages`);
+                    return;
+            }
+        },
+        undefined,
+        ctx.subscriptions
+    );
+
+    panel.webview.html = `<html>
+        <iframe name="vscode_iframe" id="mainframe" src="http://localhost:8080/" style="position:fixed; top:0; left:0; bottom:0; right:0; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999999;">
+            Your browser doesn't support iframes
+        </iframe>
+        <script>
+            const vscode = acquireVsCodeApi();
+            window.addEventListener('message', e => {
+                action = e.data.action;
+                if (action === "addPackage") {
+                    url = e.data.url
+                    version = e.data.version
+                    
+                    if (url && version) {
+                        vscode.postMessage({
+                            command: 'addSwiftPackage',
+                            url: url,
+                            version: version
+                        })
+                    } else {
+                        console.log("Unknown message: " + e)
+                    }
+                } else {
+                    console.log("Unknown message: " + e)
+                }
+            }, false);
+        </script>
+
+    </html>`;
+}
+
+export async function addPackageDependency(
+    ctx: WorkspaceContext,
+    url?: string,
+    version?: string
+): Promise<void> {
+    // The context key `swift.addPackageRefactoringAvailable` only works if the extension has been
+    // activated. As such, we also have to allow this command to run when no workspace is
+    // active. Show an error to the user if the command is unavailable.
+    if (!ctx.toolchain.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))) {
```

Comment:
What's the point of having this command available when you have no package loaded?